Implement rpc-post-v1-09 (#474)
Azure/azure-openapi-validator: docs/lro-post-final-state-via-property.md

#LROPostFinalStateViaProperty

## Category

ARM Error

## Applies to

ARM OpenAPI(swagger) specs

## Related ARM Guideline Code

- RPC-Post-V1-09

## Output Message

> A LRO POST MUST have long-running-operation-options specified and MUST have location header in the final-state-via property.

Reviewer: ```suggestion
A long running operation (LRO) post **must** have `long-running-operation-options` specified and **must** have the `final-state-via` property set to `azure-async-operation`.
```

> A long running operation (LRO) post MUST have 'long-running-operation-options' specified and MUST have the 'final-state-via' property set to 'azure-async-operation'.

Reviewer 2: >azure-async-operation [](http://example.com/codeflow?start=142&length=21)

location

## Description

> For a LRO POST long-running-operation-options must have location header in the final-state-via property.

Reviewer: ```suggestion
For long running (LRO) post operations, `long-running-operation-options` must be present and have the`final-state-via` property set to `azure-async-operation`. This indicates that the `Azure-AsyncOperation` header can be used, which points to an Operation resource.
```

## How to fix the violation

> Ensure that, for a LRO POST long-running-operation-options has location header in the final-state-via property.

Reviewer: ```suggestion
Ensure that, for long running (LRO) post operations, `long-running-operation-options` is specified and has the `final-state-via` property set to `azure-async-operation`.
```
